yang/frr-zebra.yang

```diff
@@ -0,0 +1,1887 @@
+module frr-zebra {
+  yang-version 1.1;
+  namespace "http://frrouting.org/yang/zebra";
+  prefix frr-zebra;
+
+  import ietf-yang-types {
+    prefix yang;
+  }
+
+  import ietf-inet-types {
+    prefix inet;
+  }
+
+  import frr-route-types {
+    prefix frr-route-types;
+  }
+
+  import ietf-routing-types {
+    prefix rt-types;
+  }
+
+  organization
+    "Free Range Routing";
+  contact
+    "FRR Users List: <mailto:frog@lists.frrouting.org>
+     FRR Development List: <mailto:dev@lists.frrouting.org>";
+  description
+    "This module defines a model for managing the FRR zebra daemon.";
+
+  revision 2019-06-01 {
+    description
+      "Initial revision.";
+    reference
+      "ISO/IEC 10589:2002.";
+  }
+
+  typedef unix-timestamp {
+    type uint32;
```

Comment:
I think we can use `uint64` to be prepared for the future or maybe there is a specific type for this?

Reply:
I did look around for something, and I just saw raw 'uint' values being used.

Reply:
we are using `struct timeval` which from what it looks like too me is that you are going to map that into a `uint32`?  Seems like this ( to directly translate) might be better as well.  a struct timeval is 2 uint64_t's.

Reply:
isn't there some pre-existing type for this?

also: we mostly work with monotonic clock values internally... i guess we'll be converting back & forth a lot? :open_mouth:

Reply:
Do we see any need for anything finer grained than a second? There's none I can think of, but just throwing it out there to make certain we've thought about it.

Reply:
I haven't encountered anything more fine-grained that "time_t" so far - I just was trying to indicate that some nodes have time_t semantics, and avoid using totally generic uint32. Personally, that kind of specific typing is clearer to me when I'm looking at a schema (or a datastructure).